server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -196,7 +198,8 @@ public EngineConfig(
             globalCheckpointSupplier,
             retentionLeasesSupplier,
             primaryTermSupplier,
-            tombstoneDocSupplier
+            tombstoneDocSupplier,
+            isReadOnlyReplica
```

Comment:
Segrep is feature flagged.. so maybe add a separate ctor that first checks if the feature flag is enabled and throws a `RuntimeException` if it's not?